src/misc/lv_snapshot.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file lv_snapshot.h
+ *
+ */
+
+#ifndef LV_SNAPSHOT_H
+#define LV_SNAPSHOT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include <stdint.h>
+#include <stddef.h>
+
+#include "../lv_conf_internal.h"
+#include "../core/lv_obj.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ * GLOBAL PROTOTYPES
+ **********************/
+
+/** Take snapshot for object with its children.
+ *
+ * @note The function will change the order obj in its parent.
+ *
+ * @param obj    The object to generate snapshot.
+ * @param cf     color format for generated image.
+ *
+ * @return a pointer to a image descriptor, or NULL if failed.
+ */
+lv_img_dsc_t* lv_obj_take_snapshot(lv_obj_t* obj, lv_img_cf_t cf);
+
+/** Free the snapshot image returned by @ref lv_obj_take_snapshot
+ *
+ * It will firstly free the data image takes, then the image descriptor.
+ *
+ * @param dsc    The image descriptor generated by lv_obj_take_snapshot.
+ *
+ */
+void lv_obj_snapshot_free(lv_img_dsc_t* dsc);
```

Comment:
For Micropython compatibility, if these are lv_obj methods (start with "lv_obj_...") then they must receive `lv_obj_t* obj` as their first argument, even if it's not used in the function.  

Another option is to name them differently (not start with "lv_obj_...")

Reply:
Perhaps it could be renamed to `lv_snapshot_free`?

Reply:
> Perhaps it could be renamed to `lv_snapshot_free`?

If so, lv_obj_take_snapshot should also change.

Reply:
Yes, I agree, since the file name is lv_snapshot, then the API could be lv_snapshot_take, lv_snapshot_free. I will update the code once confirmed Gablor's suggestion.